### PR TITLE
Add query execution metrics

### PR DESCRIPTION
**Pull request by ykmr1224**
### Description
- Add query execution metrics
- batch.processingTime: execution duration for batch query.
- query.execution.processingTime: Duration from start to complete query (excluding writing result)

Metrics-CloudWatch-us-west-2-10-21-2024_04_28_PM

### Related Issues
n/a

### Check List
- [ ] Updated documentation (docs/ppl-lang/README.md)
- [x] Implemented unit tests
- [ ] Implemented tests for combination with other commands
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by seankao-az — APPROVED** [review]
LGTM

File: spark-sql-application/src/main/scala/org/apache/spark/sql/JobOperator.scala [hunk 1]
```diff
@@ -195,6 +202,21 @@ case class JobOperator(
...
   }
 
+  private def emitQueryExecutionTimeMetric(startTime: Long): Unit = {
+    MetricsUtil
+      .getTimer(MetricConstants.QUERY_EXECUTION_TIME_METRIC, false)
+      .update(System.currentTimeMillis() - startTime, TimeUnit.MILLISECONDS);
+  }
+
+  private def emitBatchProcessingTimeMetric(startTime: Long): Unit = {
+    if (jobType.equalsIgnoreCase(FlintJobType.BATCH)) {
```
Comment (noCharger) [inline spark-sql-application/src/main/scala/org/apache/spark/sql/JobOperator.scala:212]: why only batch job?

File: flint-core/src/main/java/org/opensearch/flint/core/metrics/MetricConstants.java [hunk 2]
```diff
@@ -112,6 +112,16 @@ public final class MetricConstants {
...
 
+    /**
+     * Metric name for tracking the processing time of batch job.
+     */
+    public static final String BATCH_PROCESSING_TIME_METRIC = "batch.processingTime";
+
+    /**
+     * Metric for tracking the latency of query execution (start to complete query execution) excluding result write.
+     */
+    public static final String QUERY_EXECUTION_TIME_METRIC = "query.execution.processingTime";
```
Comment (noCharger) [inline flint-core/src/main/java/org/opensearch/flint/core/metrics/MetricConstants.java:123]: can we reuse STATEMENT_PROCESSING_TIME_METRIC for QUERY_EXECUTION_TIME_METRIC?

**Review by noCharger — COMMENTED** [review]
What are the dimensions of these metrics?

File: flint-core/src/main/java/org/opensearch/flint/core/metrics/MetricConstants.java (hunk 2 above)
Comment (ykmr1224) [reply flint-core/src/main/java/org/opensearch/flint/core/metrics/MetricConstants.java:123]: This metric wants to exclude the data write to result storage. (STATEMENT_PROCESSING_TIME_METRIC includes it)

File: spark-sql-application/src/main/scala/org/apache/spark/sql/JobOperator.scala (hunk 1 above)
Comment (ykmr1224) [reply spark-sql-application/src/main/scala/org/apache/spark/sql/JobOperator.scala:212]: We want to have metric for batch job. Currently there is one for statement, and streaming would have metric for each update.

**Comment by ykmr1224** [comment]
> What are the dimensions of these metrics?

Same as current ones (default ones coming from config). Might add datasource type later.

**Review by noCharger — APPROVED** [review]

**Pull request merged**